feat(angular11): upgrade to Angular 11

elenagarrone opened the pull request:
# Description

BREAKING CHANGE: update to a new major version of Angular.
This version drops support for IE9, IE10 and IE mobile.

All the spec files changed in this PR were automatically migrated by the angular cli.

https://deploy-preview-87--legal-and-general-canopy.netlify.app/

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [ ] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [ ] I have provided a story in storybook to document the changes
- [ ] I have provided documentation in the notes section of the story
- [ ] I have added any new public feature modules to public-api.ts
- [ ] I have [linked the new component](<(https://github.com/Legal-and-General/canopy/blob/master/docs/CONTRIBUTING.md#invision-dsm)>) to adobe DSM (if appropriate)

thatguynamedandy commented:
Do we know if there will be compatibility issues for applications still using Angular 10? I was wondering if it's worth using the 'next' tag to get this in and published in the interim https://semantic-release.gitbook.io/semantic-release/usage/configuration

elenagarrone commented:
> Do we know if there will be compatibility issues for applications still using Angular 10? I was wondering if it's worth using the 'next' tag to get this in and published in the interim https://semantic-release.gitbook.io/semantic-release/usage/configuration

There shouldn't be as this version is a minor one apart from the fact that it drops supports for some IE versions. It's mostly just bug fixes.

elenagarrone commented:
@thatguynamedandy I'm not 100% sure this is how you do it. I've created a `next` branch and updated both the actions and the `.releaserc`. I've changed this PR to go into `next`.

elenagarrone commented:
@thatguynamedandy could you please add the same restrictions as for `master` for the `next` branch? I don't have the correct access to do it.

owensgit reviewed (commented):
I'm new to the release process with `next`, but from what I can see in the Semantic Release docs, it looks all set correctly.

I'm just wondering if the `ci(next): add cabapility to release next branch` commit needs to come **before** the Angular 11 changes?

elenagarrone commented:
> I'm new to the release process with `next`, but from what I can see in the Semantic Release docs, it looks all set correctly.
> 
> I'm just wondering if the `ci(next): add cabapility to release next branch` commit needs to come **before** the Angular 11 changes?

Yes, I was wondering that too :/ I might create a separate PR for that just to be sure 👍

elenagarrone commented:
Remove the `next` branch commit from here and created a separate PR for it #96.

thatguynamedandy reviewed (commented):
Looks great, cheers for picking it up. I just had some questions about the versioning in package.json

elenagarrone commented on package.json:
```diff
@@ -35,27 +35,26 @@
-    "rxjs": "^6.6.2",
-    "tslib": "^2.0.3",
+    "rxjs": "^6.6.3",
+    "tslib": "^2.0.0",
```
Oh weird 🤔  I don't remember doing it, I'll have a look.

elenagarrone replied:
Done.

elenagarrone commented on package.json:
```diff
@@ -123,7 +122,7 @@
     "tslint-eslint-rules": "^5.4.0",
     "tslint-import-group-ordering": "^2.1.2",
-    "typescript": "^3.9.7",
+    "typescript": "4.0.5",
```
I really can't remember why I changed these but I updated them all now and they seem to work just fine.

elenagarrone commented on package.json:
```diff
@@ -78,7 +77,7 @@
     "@stylelint/postcss-css-in-js": "^0.37.2",
     "@stylelint/postcss-markdown": "^0.36.1",
-    "@types/jasmine": "^3.5.12",
+    "@types/jasmine": "~3.6.0",
```
Updated 👍
